templates/zerver/app/unsubscribe_stream.html

```diff
@@ -0,0 +1,16 @@
+<div id="unsubscribe_stream_modal" class="modal modal-bg hide new-style" tabindex="-1" role="dialog" aria-labelledby="unsubscribe_stream_modal_label" aria-hidden="true">
+    <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="{{ _('Close') }}"><span aria-hidden="true">&times;</span></button>
+        <h3 id="unsubscribe_stream_modal_label">{{ _("Unsubscribe from this stream") }}</h3>
+    </div>
+    <div class="modal-body">
+        <div class="content">
+            <p><strong>{{ _("Unsubscribing immediately remove you from this stream, and you will need an invitation to join again if the stream is private.")}}</strong> <br>{{_("Are you sure you want to do this?")}}</p>
+        </div>
+        <div id="unsubscribe-error"></div>
+    </div>
+    <div class="modal-footer">
+        <button class="button rounded" data-dismiss="modal">{{ _("Cancel") }}</button>
+        <button class="button rounded btn-danger" id="unsubscribe_stream_button">{{ _("Yes, Unsubscribe") }}</button>
+    </div>
+</div>
```

Comment:
This modal template file will go in folder `static/js/templates`, where you can also find other stream settings templates. It will be `handlebars` file instead of `html`.